### PR TITLE
improve the error message returned when peer verification fails

### PR DESCRIPTION
**Pull request by marten-seemann**

### CODE REVIEW

File: crypto.go [hunk 1]
```diff
@@ -93,7 +93,11 @@ func (i *Identity) ConfigForPeer(remote peer.ID) (*tls.Config, <-chan ic.PubKey)
 			return err
 		}
 		if remote != "" && !remote.MatchesPublicKey(pubKey) {
-			return errors.New("peer IDs don't match")
+			var peerID peer.ID = "(not determined)"
+			if p, err := peer.IDFromPublicKey(pubKey); err == nil {
+				peerID = p
+			}
+			return fmt.Errorf("peer IDs don't match. Expected %s, got %s", remote, peerID)
```
Comment (Stebalien) [inline crypto.go]: nit: "peer IDs don't match: expected %s, got %s" but that's mostly just preference.

File: crypto.go [hunk 2]
```diff
@@ -93,7 +93,11 @@ func (i *Identity) ConfigForPeer(remote peer.ID) (*tls.Config, <-chan ic.PubKey)
 			return err
 		}
 		if remote != "" && !remote.MatchesPublicKey(pubKey) {
-			return errors.New("peer IDs don't match")
+			var peerID peer.ID = "(not determined)"
+			if p, err := peer.IDFromPublicKey(pubKey); err == nil {
```
Comment (Stebalien) [inline crypto.go]: should we include the error?

**Review by Stebalien — APPROVED** [review]

File: crypto.go (hunk 1 above)
Comment (marten-seemann) [reply crypto.go]: Fixed.

File: crypto.go (hunk 2 above)
Comment (marten-seemann) [reply crypto.go]: Yes. Done.

**Pull request merged**